jgrapht-core/src/test/java/org/jgrapht/graph/AsSynchronizedGreaphTest.java

```diff
@@ -0,0 +1,175 @@
+package org.jgrapht.graph;
+
+import java.util.*;
+import org.jgrapht.*;
+import org.jgrapht.graph.*;
+import org.junit.*;
+
+import static org.junit.Assert.*;
+
```

Comment:
This set of tests verifies the caching logic, but does not do anything to test the actual synchronization.

We'll need a much more comprehensive set of test scenarios before this pull request can be merged.  You'll need to be creative in coming up with scenarios that have observably meaningful results while avoiding failures due to non-determinism.